api/src/main/java/jakarta/data/page/PageRequest.java

```diff
@@ -48,25 +48,24 @@
  * }
  * </pre>
  *
- * <p>A repository method will fail with a
- * {@link jakarta.data.exceptions.DataException DataException} or a more
- * specific subclass if:</p>
+ * <p>A repository method may not be declared with:</p>
```

Comment:
I don't understand why this list of user errors is split where some of them are "A repository method may not be declared with:" and others are listed as IllegalArgumentException.  We should treat all of the user errors the same and require IllegalArgumentException for all of them (except for the sorting not supported one).

Reply:
Well that was the point was trying to make earlier in the issue description: these two classes of error are _completely_ different kinds of things:

- one is an error in a method signature made by the person who wrote the repository interface, and can usually de detected at compilation time, or, at worst, at startup time, for implementations which don't do work at build time, and
- the other is an error made _by code which calls the method_, which may not even be in the same module, and which is not even in principle detectable at compilation time.

Bundling these two kinds of failure into the same list is a category error.

One of them is an error in the definition of the repository (assigned `UnsupportedOperationException` elsewhere in the spec), and one of them is an error in the arguments passed to the repository method (a classic example of an `IllegalArgumentException`).

Reply:
You're calling them both "user error", but there are two kinds of users here:

- the user of Jakarta Data (the person who creates repositories), and
- the user of the repository, who maybe doesn't know anything about Jakarta Data.

Reply:
Okay, I see what distinction you are making here. I got confused because for one category (invalid method signature) it doesn't state what the exception is and for some reason I assumed the JavaDoc said it was IllegalArgumentException as well, but I can see now you didn't write that and left it undefined in both JavaDoc and spec as to what the exception. I suppose this is fine. The user has been informed to avoid the invalid combinations of parameters, which is the part that really matters here.

Reply:
Yeah, I don't want to state that in Javadoc because the user should definitely not be writing code which depends on that exception type.